src/components/LayoutTree.tsx

```diff
@@ -446,6 +452,55 @@ const TreeNode = ({
     return lastPatches?.some(patch => deleteActions.includes(patch.actions[0]))
   })
 
+  const [showLineEndFauxCaret, setShowLineEndFauxCaret] = useState(false)
+  const [showLineStartFauxCaret, setShowLineStartFauxCaret] = useState(false)
+
+  // Hide the faux caret when typing occurs.
+  editingValueStore.subscribe(() => {
+    if (isTouch && isSafari() && caretRef.current) {
+      caretRef.current.style.display = 'none'
+      setShowLineStartFauxCaret(false)
+      setShowLineEndFauxCaret(false)
+    }
+  })
+
+  // If the thought isCursor and edit mode is on, position the faux cursor at the point where the
+  // selection is created.
+  useEffect(() => {
```

Comment:
I think it would be a good idea to factor the bulk of this logic out into a custom hook. It would be nice to put it all in a custom component, but `showLineStartFauxCaret` and `showLineEndFauxCaret` are used in `TreeNode` itself, so it seems like a custom hook is the best approach. I would suggest stuffing `showLineStartFauxCaret` and `showLineEndFauxCaret` into the hook and having it return an opaque style object since `TreeNode` doesn't really need to know the implementation details as long as `--faux-caret-line-start-opacity` and `--faux-caret-line-end-opacity` are applied correctly. Let me know what you think.